common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -552,6 +553,76 @@ public ICacheRecord load(@NonNull final String clientId,
                 null // not applicable
         );
 
+        {
+            // If we didn't find an RT in the cache, this could be a "TSL-seed" or "dual-client stack"
+            // scenario
+            //
+            // Defining these terms:
+            // TSL-seed: another 1P TSL integrated app has put a token into our cache so we can
+            //     pick it up
+            //
+            // Dual-Client stack: two FoCI-enabled app registrations are sharing a single binary
+            //     and accordingly, can share RTs.
+            //     Examples for this might be TFL/TFW - which uses multiple client ids to enable
+            //     different scenarios depending on enterprise vs. consumer usage
+
+            // Unlike the broker, where we check if an app is FoCI prior to making a network call
+            // with an arbitrary FoCI RT we find in the cache, if we're in standalone mode and find
+            // a FoCI RT in the cache, the current app must also be FoCI (!!!)
+            //
+            // Making the assumption that the current client id can use any FoCI RT we find in the
+            // cache is strictly contingent that app developers NOT mix 1P & 3P registrations into
+            // the same binary. If you do this, Bad Things will happen and you'll get confusing
+            // errors that the RT used doesn't match the client app registration. Also, this
+            // assumption means we don't need to implement "FoCI probing" and/or track FoCI app meta
+
+            if (refreshTokens.isEmpty()) {
+                // Look for an arbitrary RT matching the current user.
+                // If we find one, check that it is FoCI, if it is, assume it works.
+                final List<Credential> fallbackRts = mAccountCredentialCache.getCredentialsFilteredBy(
+                        account.getHomeAccountId(),
+                        account.getEnvironment(),
+                        CredentialType.RefreshToken,
+                        null, // wildcard (*)
+                        isMultiResourceCapable
+                                ? null // wildcard (*)
+                                : account.getRealm(),
+                        isMultiResourceCapable
+                                ? null // wildcard (*)
+                                : target,
+                        null // not applicable
+                );
+
+                if (!fallbackRts.isEmpty()) {
+                    Logger.verbose(
+                            TAG + methodName,
+                            "Inspecting fallback RTs for a FoCI match."
+                    );
+
+                    // Any arbitrary RT should be OK -- if multiple clients are stacked, they're either
```

Comment:
Are there any (or could there be any) first party apps that are not part of FoCI for some reason?

Reply:
Yes, we do have 1P apps that are not FoCI. They won't benefit from these changes

Had an offline sync w/ @shoatman -- the output of our discussion was that any 1P app that plans to dual-client stack _must_ have both be FoCI, or neither. We won't support a 'blended' model where one is and the other is not